source/core/core.natvis

```diff
@@ -62,4 +62,45 @@
       <ExpandedItem>pointer</ExpandedItem>
     </Expand>
 </Type>
+
+
+<Type Name="Slang::Safe32Ptr&lt;*&gt;">
+    <Expand>
+        <ExpandedItem>($T1*)(m_base->m_data + m_offset)</ExpandedItem>
+    </Expand>
+</Type>
+
+<Type Name="Slang::Relative32Ptr&lt;*&gt;">
+    <Expand>
+        <ExpandedItem>(m_offset == 0x80000000) ? nullptr : ($T1*)(((char*)this) + m_offset)</ExpandedItem>
```

Comment:
I haven't gotten to the implementation yet, but why not just use zero as the sentinel value here (which means a pointer can't point to itself). Any other bit pattern has the potential for there to be some address that can't be referenced, right?

I guess there is also the language-lawyer question of whether it is safer here to do the math on `uintptr_t` first and then convert to the pointer rather than add `m_offset` directly to `this` (which could technically overflow/underflow, and I don't remember the semantic guarantees of overflow/underflow for pointer).

Reply:
Although perhaps somewhat unusual, but 0 is avoided because I could have a pointer that pointed to a struct where the pointer was the first member, and then the offset would be 0.

As pointed out later the idea here is to use relative pointers of fixed size for serializability purposes. This is a 32 bit version.